.github/actions/setup/action.yml

```diff
@@ -0,0 +1,90 @@
+name:
+  Setup
+description:
+  Setup of the needed software
+
+inputs:
+  os:
+    description: The os for the run.
+    required: true
+
+  cxx:
+    description: The c++ compiler to use.
+    required: true
+
+  cxx-standard:
+    description: The c++ standard to use.
+    required: true
+
+runs:
```

Comment:
I would probably add this to the main workflow since it's only used there.  It would sidestep the dependabot thing, too.

Reply:
I had in mind that it can be used it in the other `cmake` workflows as well as they are doing similar/the same things to set up the environment.

But I'm also fine with moving the action into the workflow itself.